high_card.rb

```diff
@@ -1,5 +1,39 @@
 # Use these two arrays to generate a deck of cards.
 ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
-
 players = []
+deck = []
+player_input = ""
+
+ranks.each do |rank|
+  suits.each do |suit|
+    deck.push([rank,suit])
+  end
+end
+deck = deck.shuffle
+
+until(player_input=="play")
+  puts("#{players.length} players so far. Enter a player name, or type 'play':")
+  player_input = gets.chomp.to_s
+  players.push(player_input)
+end
+
+players.pop
+
+
+hand = []
+
+
+players.each do |player|
+  hand.push(deck[player.to_i])
+  deck.shift
```

Comment:
`deck.shift` removes an item from the beginning of the `deck` array. Even though it doesn't have a `!`, it does actually modify `deck`. This is why at the end you're left with only 50 cards -- the other 2 have essentially vanished. To "capture" them, you'd do something like:

```rb
dealt_card = deck.shift
```

`.shift` returns the item that was removed from the array.